### PR TITLE
init ro sidecar mount

### PR DESCRIPTION
**Pull request by ransomw1c**
toward #135

### CODE REVIEW

**Comment by ransomw1c** [comment]
opening this to pivot back to making the mutable filesystem available from the CLI.

as is, it demonstrates mounting the read-only filesystem from a k8s sidecar container.  the `make fuse-demo-ro` target builds docker images for both a datamon container as well as an additional container, deploys the images to kubernetes, and starts a shell in the container that doesn't contain datamon.  from that shell, `ls /tmp/mount` displays the contents of a bundle via a FUSE mount in the sidecar container.

File: hack/k8s/example-ro.template.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: datamon-ro-demo
+    spec:
+      containers:
+      - name: demo-shell
+        image: fuse-demo-shell:latest
+        imagePullPolicy: "Never"
```
Comment (kerneltime) [inline hack/k8s/example-ro.template.yaml]: Why never?

File: hack/k8s/example-ro.template.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        tty: true
+        volumeMounts:
+        - mountPath: /tmp/mount
+          name: fuse-mountpoint
+          mountPropagation: "HostToContainer"
+      - name: datamon-sidecar
+        image: fuse-demo-sidecar:latest
+        imagePullPolicy: "Never"
+        command: ["datamon"]
+        args: ["bundle", "mount", "--bundle", "1Kbtk5Mo0M8ck2lcnuhMNYiV9DO", "--repo", "ransom-datamon-test-repo", "--destination", "/tmp", "--mount", "/tmp/mount"]
```
Comment (kerneltime) [inline hack/k8s/example-ro.template.yaml]: skip the hash and let it download the latest for demo

File: hack/k8s/example-ro.template.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        securityContext:
+          privileged: true
```
Comment (kerneltime) [inline hack/k8s/example-ro.template.yaml]: What is the minimum securityContext? Does SYS_ADMIN work?

File: hack/k8s/example-ro.template.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        stdin: true
+        tty: true
+        volumeMounts:
+        - mountPath: /tmp/mount
+          name: fuse-mountpoint
+          mountPropagation: "Bidirectional"
```
Comment (kerneltime) [inline hack/k8s/example-ro.template.yaml]: Do we need bidirectional for within pod sharing?

Comment (ransomw1c) [reply hack/k8s/example-ro.template.yaml]: yes, i have tried toggling this to [other strings](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) and re-running the `fuse-demo/` scripts to drop into the `demo-shell`, and `ls /tmp/mount` from the demo shell only displays the demo files with bidirectional mount propagation.

in particular, what was not obvious to me (at the level of k8s abstraction) is why there's host->container unidirectional propagation and not container->host.

File: hack/k8s/example-ro.template.yaml (hunk 3 above)
Comment (ransomw1c) [reply hack/k8s/example-ro.template.yaml]: i've tried both

```
        securityContext:
          allowPrivilegeEscalation: true
          capabilities:
            add:
              - SYS_ADMIN
```
and
```
        securityContext:
          capabilities:
            add:
              - SYS_ADMIN
```

and neither allows access to fuse

Comment (kerneltime) [reply hack/k8s/example-ro.template.yaml]: ok

File: hack/k8s/example-ro.template.yaml (hunk 4 above)
Comment (kerneltime) [reply hack/k8s/example-ro.template.yaml]: I guess the mounts need to go back to the os for it to share with the other container cause pod is a k8s abstraction and as far as the OS is concerned these are separate containers.

File: hack/k8s/example-ro.template.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: datamon-ro-demo
+    spec:
+      containers:
+      - name: demo-shell
+        image: fuse-demo-shell:latest
+        imagePullPolicy: "IfNotPresent"
```
Comment (kerneltime) [inline hack/k8s/example-ro.template.yaml]: I think this is best set to always? From the documentation ```imagePullPolicy is omitted and either the image tag is :latest or it is omitted: Always is applied.``` so the policy here will be ignored cause we want to pick the latest?

**Pull request merged**